.github/workflows/spelling.yml

```diff
@@ -13,21 +13,40 @@ on:
     types: ['opened', 'reopened', 'synchronize']
 
 jobs:
-  build:
+  spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
+      actions: read
+    outputs:
+      followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-20.04
+    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    concurrency:
+      group: spelling-${{ github.event.pull_request.number || github.ref }}
+      # note: If you use only_check_changed_files, you do not want cancel-in-progress
+      cancel-in-progress: true
     steps:
-    - name: checkout-merge
-      if: "contains(github.event_name, 'pull_request')"
-      uses: actions/checkout@v2.0.0
+    - name: check-spelling
+      id: spelling
+      uses: check-spelling/check-spelling@v0.0.20
       with:
-        ref: refs/pull/${{github.event.pull_request.number}}/merge
-        fetch-depth: 5
-    - name: checkout
-      if: "!contains(github.event_name, 'pull_request')"
-      uses: actions/checkout@v2.0.0
-      with:
-        fetch-depth: 5
-    - uses: check-spelling/check-spelling@v0.0.19
+        suppress_push_for_open_pull_request: 1
+        checkout: true
+        post_comment: 0
+
+  comment:
+    name: Report
+    runs-on: ubuntu-latest
+    needs: spelling
+    permissions:
+      contents: write
```

Comment:
Why does the spelling action need `contents` permissions?

Reply:
GitHub treats `comment`ing on a commit as a `write` operation (on the repository contents). And `comment`ing on a pul request as a write operation (on pull requests).

It's an unfortunate detail.

But note that the default permissions on most organizations is roughly `*: write`, so this is actually a reduction in the scope permissions of the workflow.